src/main/scala/org/broadinstitute/dsde/firecloud/core/ProfileClient.scala

```diff
@@ -273,4 +283,64 @@ class ProfileClientActor(requestContext: RequestContext) extends Actor with Fire
       postResponse.entity.asString.contains(Conflict.reason)
   }
 
+  def syncWhitelist: Unit = {
+    val (bucket, file) = (FireCloudConfig.Nih.whitelistBucket, FireCloudConfig.Nih.whitelistFile)
+    val whitelist = Source.fromInputStream(HttpGoogleServicesDAO.getBucketObjectAsInputStream(bucket, file)).getLines().toSet
+
+    // note: everything after this point involves Futures
+
+    val nihEnabledFireCloudUsers = NIHWhitelistUtils.getCurrentNihUsernameMap map { mapping =>
+      mapping.collect { case (fcUser, nihUser) if whitelist contains nihUser => fcUser }.toSeq
+    }
+
+    val memberList = nihEnabledFireCloudUsers map { subjectIds => RawlsGroupMemberList(userSubjectIds = Some(subjectIds))}
+
+    val pipeline = addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken)) ~> sendReceive
+    val url = FireCloudConfig.Rawls.overwriteGroupMembershipUrlFromGroupName(FireCloudConfig.Nih.rawlsGroupName)
+
+    memberList flatMap { members => pipeline { Put(url, members) } }
+  }
+
 }
+
+object NIHWhitelistUtils {
+  def getAllUserValuesForKey(key: String)(implicit arf: ActorRefFactory, ec: ExecutionContext): Future[Map[String, String]] = {
+    val pipeline = addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken)) ~> sendReceive
```

Comment:
future refactoring ... this code:
```
addCredentials(OAuth2BearerToken(HttpGoogleServicesDAO.getAdminUserAccessToken))
```
could/should be encapsulated into a custom RequestTransformer in FireCloudRequestBuilding, something like "addAdminCredentials"